src/v/serde/thrift/compact.cc

```diff
@@ -81,6 +81,15 @@ bytes encode_string(std::string_view str) {
     return b;
 }
 
+iobuf encode_binary(iobuf b) {
+    auto len = unsigned_vint::to_bytes(static_cast<int64_t>(b.size_bytes()));
```

Comment:
why the cast to `int64_t` is required here ? The method accepts `uint32_t`

Reply:
Accidental - good catch. removed